strapi/strapi
# refactor: admin build pipeline

### What does it do?

* removes chunk names because they won't work with vite
* packs-up part of the admin package
* moves build & develop command to the admin package and is then injected into the strapi package
* refactors the pipeline to use a common build context (makes it easier to add new builders quickly)
* integrates webpack-dev-middleware & webpack-hot-middleware so development is all done in one command – `yarn develop` in an example
* ensures there are aliases for the local packages if we're in the strapi monorepo
* adds a global logger, cwd & strapi instance (unloaded) to the Strapi CLI context to be added to all commands
* Adds a runtime directory to strapi (no more .cache folder – could solve pnpm issues)
* adds contributor documentation

### Why is it needed?

* Preps the way for to introduce vite
* Better DX for working with strapi
* CLI commands can now use the same logger 😎 
* Runtime directory makes the whole thing simpler meaning we could make more of a `strapi.config.ts` if we wanted too..
* Admin could(?) be rendered without a the server being present with some more work inside another app - e.g nextjs 

### How to test it?

* run the project locally & use `yarn develop` in an example, you should be able to continue working as normal

### Related issue(s)/PR(s)

* resolves CONTENT-1624
* resolves CONTENT-1631
* resolves CONTENT-1629
* resolves CONTENT-1628
